crates/pallets/starknet/src/lib.rs

```diff
@@ -753,48 +753,78 @@ pub mod pallet {
                     let deploy_account_transaction = transaction
                         .clone()
                         .from_deploy(&Self::chain_id_str())
                         .map_err(|_| TransactionValidityError::Unknown(UnknownTransaction::CannotLookup))?;
-                    // Pallet::<T>::validate_tx(deploy_account_transaction, TxType::DeployAccount)?;
                     ValidTransaction::with_tag_prefix("starknet")
                         .priority(u64::MAX - (TryInto::<u64>::try_into(transaction.nonce)).unwrap())
-                        .and_provides((deploy_account_transaction.sender_address, transaction.nonce))
-                        .longevity(T::TransactionLongevity::get())
-                        .propagate(true)
-                        .build()
+                        // This is a transaction identifier for substrate
+                        .and_provides(vec![deploy_account_transaction.sender_address, transaction.nonce])
+                        .longevity(64_u64)
+                        .propagate(true).build()
+                }
+                Call::consume_l1_message { transaction } => {
+                    // Message consumptions don't go through an account contract so no need to identify them with an id.
+                    let tx = ValidTransaction::with_tag_prefix("starknet")
+                        .priority(u64::MAX - (TryInto::<u64>::try_into(transaction.nonce)).unwrap())
+                        .and_provides(vec![Felt252Wrapper::default(), transaction.nonce])
+                        .longevity(64_u64)
+                        .propagate(true);
+
+                    tx.build()
                 }
-                Call::consume_l1_message { transaction } => ValidTransaction::with_tag_prefix("starknet")
-                    .priority(u64::MAX - (TryInto::<u64>::try_into(transaction.nonce)).unwrap())
-                    .and_provides((transaction.sender_address, transaction.nonce))
-                    .longevity(T::TransactionLongevity::get())
-                    .propagate(true)
-                    .build(),
                 _ => InvalidTransaction::Call.into(),
             }
         }
+
+        /// From substrate documentation:
+        /// Validate the call right before dispatch.
+        /// This method should be used to prevent transactions already in the pool
+        /// (i.e. passing validate_unsigned) from being included in blocks in case
+        /// they became invalid since being added to the pool.
+        ///
+        /// In the default implementation of pre_dispatch for the ValidateUnsigned trait,
+        /// this function calls the validate_unsigned function in order to verify validity
+        /// before dispatch. In our case, we validate that the transaction is valid for
+        /// the current block by calling `Transaction::validate_tx`.
+        fn pre_dispatch(call: &Self::Call) -> Result<(), TransactionValidityError> {
+            match call {
```

Comment:
This is the main change and should be discussed: in order to accelerate transactions processing by the transaction pool, I removed transaction validation from `validate_unsigned` and moved it to `pre_dispatch`. This could cause the pool to accept transactions which would later be rejected, hence having useless transactions in the pool. Moving validation to `validate_unsigned` causes too big a workload on the pool, which causes it to lose time validating ALL transactions (which is an action that is awaited, see [here](https://github.com/paritytech/substrate/blob/master/client/transaction-pool/src/graph/pool.rs#L155) and [here](https://github.com/paritytech/substrate/blob/master/client/transaction-pool/src/graph/pool.rs#L374)), and prevents the pool from pruning executed transactions, causing reexecution of transactions and hence nonce errors.